### PR TITLE
feat: add feature client for izanami

### PR DESCRIPTION
**Pull request by sguiheux**

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package izanami_client
+
+type client struct {
```
Comment (fsamin) [inline client.go]: should be public 
or make a interface and only return it

File: client.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+	clientID     string
+	clientSecret string
+}
+
+type featureClient struct {
+	client *client
+}
+
+// New creates a new izanami client
+func New(host, clientID, secret string) *client {
```
Comment (fsamin) [inline client.go]: is it possible to ping the izanami and raise an error if not available, or auth failed ?

File: client.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+	return &client{
+		hostname:     host,
+		clientID:     clientID,
+		clientSecret: secret,
+	}
+}
+
+// Feature creates a specific client for feature management
+func (c *client) Feature() *featureClient {
```
Comment (fsamin) [inline client.go]: should return a public interface

File: client.go [hunk 4]
```diff
@@ -0,0 +1,27 @@
+package izanami_client
```
Comment (fsamin) [inline client.go]: package should not contains underscore

File: feature.go [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+		return features, errListAll
+	}
+
+	if err := json.Unmarshal(res, &features); err != nil {
+		return features, err
+	}
+	return features, nil
+}
+
+func (c *featureClient) listAll() ([]FeatureModel, error) {
```
Comment (fsamin) [inline feature.go]: should be public ?

File: feature.go [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+		features = append(features, res.Results...)
+		if res.Metadata.Page == res.Metadata.PageSize {
+			break
+		}
+
+	}
+	return features, nil
+}
+
+func (c *featureClient) create(feat FeatureModel) error {
```
Comment (fsamin) [inline feature.go]: should be public ?

**Review by fsamin — CHANGES_REQUESTED** [review]

File: feature.go (hunk 6 above)
Comment (sguiheux) [reply feature.go]: done

File: feature.go (hunk 5 above)
Comment (sguiheux) [reply feature.go]: done

File: client.go (hunk 4 above)
Comment (sguiheux) [reply client.go]: done

File: http.go [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+const (
+	httpParamPage     = "page"
+	httpParamPageSize = "pageSize"
+
+	headerClientID     = "Izanami-Client-Id"
+	headerClientSecret = "Izanami-Client-Secret"
+)
+
+var (
+	httpClient = &http.Client{
```
Comment (fsamin) [inline http.go]: should be a public attribute of the client

File: README.md [hunk 8]
```diff
@@ -1,2 +1,43 @@
...
+Go client for [izanami](https://github.com/maif/izanami)
+
+
+### Usage
+
+```
+c := New("host", "clientID", "clientSecret")
+	
+// List all features
+features, errF := c.Feature().listAll()
```
Comment (fsamin) [inline README.md]: update the doc

**Review by fsamin — CHANGES_REQUESTED** [review]

**Comment by fsamin** [comment]
we should wrote a very few tests, to able able to check if the server new versions doesn't have impact on this code

File: README.md [hunk 9]
```diff
@@ -1,2 +1,43 @@
+# [WORK IN PROGRESS]
+
 # izanami-go-client
-Go client for izanami
+Go client for [izanami](https://github.com/maif/izanami)
+
+
+### Usage
+
+```
```
Comment (bnjjj) [inline README.md]: add go

**Review by bnjjj — CHANGES_REQUESTED** [review]

**Review by fsamin — APPROVED** [review]

**Review by bnjjj — APPROVED** [review]

**Review by yesnault — APPROVED** [review]

**Pull request merged**